src/hooks/useEncryptedVotesEvents.ts

```diff
@@ -15,6 +16,12 @@ export default function useEncryptedVotesEvents(
 ) {
   const { addError } = useContext(ErrorContext);
 
+  const previousAddress = usePrevious(address);
+  // To prevent a "bad MAC" error, try not to run the query if the previousAddress is defined but it doesn't equal current address.
+  // This happens because the wallet state is async -- sometimes when disconnecting, address is defined before we change the state.
+  const addressCheck = !(
+    previousAddress !== null && previousAddress !== address
+  );
```

Comment:
im not sure i follow this logic, are you sure it works?  so it will query if:
1. previousAddress == null
2. previousAddress == address
for 1. its probably fine, but is it the case that 2 will ever be true, maybe i dont understand what previousAddress represents

Reply:
So sometimes we will have the address (only when. they've connected) but they've *changed* addresses.

This leads to the Bad MAC error. Basically, as other parts of the app disconnect them as they change address, I don't want this race condition to show them an error which shouldn't concern them.